### DISCUSSION TITLE
how to manage different inventory groups

### DISCUSSION THREAD

**Discussion by hmoffatt** [semaphoreui/semaphore#1526] [Ideas]
I have two sets of hosts, test and production. Currently they're in one inventory, and added to appropriate groups. I'd like to define task templates to run various tasks on either test or production or both.

Semaphore doesn't make it easy to run a playbook against just one group; you have to add the ansible --limit parameter to the CLI arguments manually. It would be nice if when defining an inventory, you could also define a group to limit to.

Alternatively I could split my inventory into multiple files and use a Semaphore inventory for each of them, but then there is no way to specify multiple files in one Semaphore inventory so I couldn't run tasks against both groups at once.

Does anyone have any better ideas on how to deal with this except continue to use --limit manually?

**Comment by rogertheshruber** [comment]
use 3 files, one for prod one for test and one for all.

**Reply by hmoffatt** [reply]
I have multiple host groups (apps, database, etc) as well as multiple stages (prod, test) and I might want to run against all or all prod or test database etc. I'd prefer not to repeat hosts in the inventory files.

**Comment by rogertheshruber** [comment]
it looks like to can add multiple inventory files to ansible-playbook by using "-i inventoryFile1 -i inventoryFile2".  if you go wih static file inventory you could select one inventory in the template and add one to the command line with something like this.

[
  "-i",
  "/path/inventoryFile"
]

**Reply by hmoffatt** [reply]
That would work but it's not really any less complex than using --limit.

**Comment by MSandro** [comment]
I've opened a similar discussion #1613 
I hope there will be some improvements. It would also be helpful to set up several inventories in a task template.

**Comment by tboerger** [comment]
Moving this to ideas category as this should be used as a base to restructure the handling of groups.

**Comment by ignis-draco** [comment]
I would also very much welcome it if you could assign several inventories to a task (both static and dynamic). It would also be a great advantage if you could assign several user credentials to the inventories.

**Comment by sickwell6988** [comment]
@hmoffatt @tboerger @ignis-draco @MSandro @rogertheshruber 
Guys, I red carefully all the thread, and have the solution:

Just provide it like --limit and your_hosts. See below.

![image](https://github.com/user-attachments/assets/4ba63df5-75e4-409c-99ff-9eb4529bc839)

Semaphore UI build: v2.12.14-685c726-1739570731

**Reply by hmoffatt** [reply]
Yes of course this is possible, I am using it, but it's not a good interface.

**Comment by Zeashh** [comment]
I'm looking for a lighter AWX/Ansible Tower alternative and have found this to be one of the few things missing from Semaphore. Could we perhaps see an option to select multiple inventories in task templates?
